tool/jt.rb

```diff
@@ -968,7 +967,14 @@ def test_ecosystem(*args)
     single_test            = !args.empty?
     test_names             = single_test ? '{' + args.join(',') + '}' : '*'
 
-    success = Dir["#{tests_path}/#{test_names}.sh"].sort.all? do |test_script|
+    candidates = Dir["#{tests_path}/#{test_names}.sh"].sort
+    if candidates.empty?
+      targets = Dir["#{tests_path}/*.sh"].sort.map { |f| File.basename(f).split(".")[0] }
```

Comment:
Could you use
```ruby
File.basename(f, ".*")
```
here?
It achieves the safe but is safer in case there is a dot in the filename.